src/arraymancer/linear_algebra/special_matrices.nim

```diff
@@ -47,3 +47,137 @@ proc vandermonde*[T](x: Tensor[T], order: int): Tensor[float] =
   let orders = arange(order.float)
   for i, ax in enumerateAxis(result, axis = 1):
     result[_, i] = (x ^. orders[i]).unsqueeze(axis = 1)
+
+proc diag*[T](d: Tensor[T], k = 0): Tensor[T] {.noInit.}=
```

Comment:
```suggestion
proc diag*[T](d: Tensor[T], k = 0): Tensor[T] {.noInit.} =
```

Reply:
You are right, implementing the diag procedure based on set_diagonal makes more sense as it reduces code duplication and let's us easily add support for the anti flag as well.
I will fix the missing spaces as well.